### PR TITLE
Removed outdated statements about asset_usage

### PR DESCRIPTION
**Pull request by brianreavis**
Incorporates #991 and includes additional updates to address outdated information.

### CODE REVIEW

**Comment by afonsolage** [comment]
The main branch already has this modifications, I added this on #991. Or did I miss something?

**Comment by brianreavis** [comment]
> - Asset loaders (GLTF, etc) now load meshes and textures without `asset_usage`. These assets will be removed from `Assets<Mesh>` and `Assets<Image>` once `RenderAssets<Mesh>` and `RenderAssets<Image>` contain the GPU versions of these assets, in order to reduce memory usage. If you require access to the asset data from the CPU in future frames after the GLTF asset has been loaded, modify all dependent `Mesh` and `Image` assets and set `asset_usage` to `RenderAssetUsages::default()`, which implies `RenderAssetUsage::MAIN_WORLD | RenderAssetUsage::RENDER_WORLD`.

This statement is inaccurate for two reasons:

1. Assets aren't loaded by default anymore (https://github.com/bevyengine/bevy/pull/11212)
2. "Asset loaders (GLTF, etc) now load meshes and textures without `asset_usage`" doesn't really make sense.

@alice-i-cecile Can you reopen this?

**Comment by alice-i-cecile** [comment]
Ah, got it. Can you resolve merge conflicts so we can see the true diff?

**Comment by brianreavis** [comment]
All set

**Review by alice-i-cecile — APPROVED** [review]
Ah much clearer now. Thanks!

File: content/learn/migration-guides/0.12-to-0.13.md [hunk 1]
```diff
@@ -1231,7 +1232,7 @@ The increased `Exposure::ev100` means that all existing 3D lighting will need to
   - The trait now requires `Clone`.
   - The `ExtractedAsset` associated type has been removed (the type itself is now extracted).
   - The signature of `prepare_asset()` is slightly different
-  - A new `asset_usage()` method is now required (return `RenderAssetUsage::RENDER_WORLD` to match the previous behavior).
+  - A new `asset_usage()` method is now required (return `RenderAssetUsages::default()` to match the previous behavior).
```
Comment (afonsolage) [inline content/learn/migration-guides/0.12-to-0.13.md:1235]: The original migration guide stated the `asset_usage` should return `RenderAssetPersistencePolicy::Unload` to match the previous `AssetRender` behavior, which maps to `RenderAssetUsage::RENDER_WORLD` and then unloading the asset from CPU.

By changing it to `RenderAssetUssages::default()` the asset will be kept both in CPU and GPU. This was a mistake on original PR or something has changed in another PR?

**Review by afonsolage — CHANGES_REQUESTED** [review]

File: content/learn/migration-guides/0.12-to-0.13.md (hunk 1 above)
Comment (rparrett) [reply content/learn/migration-guides/0.12-to-0.13.md:1235]: Seems like a mistake in the original guide to me.

**Review by afonsolage — APPROVED** [review]

**Pull request merged**